lib/src/layout/sidebar_item.dart

```diff
@@ -0,0 +1,502 @@
+  ///
+  /// Defaults to transparent.
+  final Color? unselectedColor;
+
+  /// The color to paint the disclosure item.
+  final Color? disclosureColor;
+
+  /// The [shape] property specifies the outline (border) of the
+  /// decoration. The shape must not be null. It's used alonside
+  /// [selectedColor].
+  final ShapeBorder? shape;
+
+  List<SidebarItem> get _allItems {
+    List<SidebarItem> result = [];
+    items.forEach((element) {
+      if (element.disclosureItems != null) {
+        result.addAll(element.disclosureItems!);
+      } else {
+        result.add(element);
+      }
+    });
+    return result;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (items.isEmpty) return SizedBox.shrink();
+    assert(debugCheckHasMacosTheme(context));
+    assert(currentIndex < _allItems.length);
+    final theme = MacosTheme.of(context);
+    return IconTheme.merge(
+      data: const IconThemeData(size: 20),
+      child: _SidebarItemsConfiguration(
+        selectedColor: selectedColor ?? theme.primaryColor,
+        unselectedColor: unselectedColor ?? _transparent,
+        disclosureColor: disclosureColor ?? _defaultDisclosureColor,
+        shape: shape ?? _defaultShape,
+        child: ListView(
+          controller: scrollController,
+          physics: const ClampingScrollPhysics(),
+          padding: EdgeInsets.all(10.0 - theme.visualDensity.horizontal),
+          children: List.generate(items.length, (index) {
+            final item = items[index];
+            if (item.disclosureItems != null) {
+              return _DisclosureSidebarItem(
+                item: item,
+                selectedItem: _allItems[currentIndex],
+                onChanged: (item) {
+                  onChanged(_allItems.indexOf(item));
+                },
+              );
+            }
+            return _SidebarItem(
+              item: item,
+              selected: _allItems[currentIndex] == item,
+              onClick: () => onChanged(_allItems.indexOf(item)),
+            );
+          }),
+        ),
+      ),
+    );
+  }
+}
+
+class _SidebarItemsConfiguration extends InheritedWidget {
+  const _SidebarItemsConfiguration({
+    Key? key,
+    required this.child,
+    this.selectedColor = _transparent,
+    this.unselectedColor = _transparent,
+    this.disclosureColor = _defaultDisclosureColor,
+    this.shape = _defaultShape,
+  }) : super(key: key, child: child);
+
+  final Widget child;
+
+  final Color selectedColor;
+  final Color unselectedColor;
+  final Color disclosureColor;
+  final ShapeBorder shape;
+
+  static _SidebarItemsConfiguration of(BuildContext context) {
+    return context
+        .dependOnInheritedWidgetOfExactType<_SidebarItemsConfiguration>()!;
+  }
+
+  @override
+  bool updateShouldNotify(_SidebarItemsConfiguration oldWidget) {
+    return true;
+  }
+}
+
+/// A macOS style navigation-list item intended for use in a [Sidebar]
+class _SidebarItem extends StatelessWidget {
```

Comment:
Why private? There are some times when a user might want to use a SidebarItem directly

Reply:
When? I don't see a reason for him wanting to use it outside of `SidebarItems`. This widget is also dependent of `SidebarItems`

Reply:
I often need a "ListTile" sort of widget that is of the macOS variety; this seems like a decent candidate

Reply:
A different widget should be created for this purpose then.